webclient/app/src/app/pages/appEditor/sections/erSection/ErSection.jsx

```diff
@@ -63,6 +63,15 @@ function ErDesigner(props) {
             return;
         }
 
+        let newEntities = [...entities];
+        const locallyFoundIndex = newEntities.findIndex(entity => entity.id === entityId);
+        if (newEntities[locallyFoundIndex].isNewEntity) {
+            return new Promise(resolve => {
```

Comment:
```suggestion
            // Returns a promise so that there is no difference in the data type between a server deleted entity vs a locally deleted item
            return new Promise(resolve => {
```